src/index.ts

```diff
@@ -122,10 +127,22 @@ export function kebabCase<
         .join(joiner ?? "-") as JoinByCase<T, Joiner>);
 }
 
+export type SnakeCase<T extends string | readonly string[]> = JoinByCase<
+  T,
+  "_"
+>;
+
 export function snakeCase(): "";
 export function snakeCase<T extends string | readonly string[]>(
   string_: T
-): JoinByCase<T, "_">;
+): SnakeCase<T>;
 export function snakeCase<T extends string | readonly string[]>(string_?: T) {
-  return kebabCase(string_ || "", "_") as JoinByCase<T, "_">;
+  return kebabCase(string_ || "", "_") as SnakeCase<T>;
 }
+
+export {
+  type CamelCase,
+  type JoinByCase,
+  type PascalCase,
+  type SplitByCase,
+} from "./types";
```

Comment:
Since an internal implementation might change, would it make sense to expose camel/pascal type utils for now or do you see an external usage for split/join?